src/main/java/online/talkandtravel/service/impl/OnlineServiceImpl.java

```diff
@@ -0,0 +1,152 @@
+package online.talkandtravel.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import online.talkandtravel.exception.user.UserNotFoundException;
+import online.talkandtravel.model.dto.user.OnlineStatusDto;
+import online.talkandtravel.model.entity.User;
+import online.talkandtravel.repository.UserRepository;
+import online.talkandtravel.service.OnlineService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static online.talkandtravel.util.AuthenticationUtils.getUserFromPrincipal;
+import static online.talkandtravel.util.RedisUtils.*;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class OnlineServiceImpl implements OnlineService {
+
+    private final RedisTemplate<String, String> redisTemplate;
+    private final SimpMessagingTemplate messagingTemplate;
+    private final UserRepository userRepository;
+
+    public static final String USERS_ONLINE_STATUS_ENDPOINT = "/users/onlineStatus";
```

Comment:
i think this constant should be moved, but where?)) 'ApiPathConstants'? 'WebSocketConfig'? or ...?